libs/components/src/next/form/checkbox/vl-checkbox.component.ts

```diff
@@ -0,0 +1,161 @@
+            if (this.checked) {
+                this.setValue(this.value || 'true');
+            } else {
+                this.setValue('');
+            }
+            if (this.touched || this.initialCheckedValue !== this.checked) {
+                this.checkValidity();
+            }
+            if (!this.touched) {
+                this.touched = true;
+            }
+        }
+    }
+
+    render(): TemplateResult {
+        return html` ${!this.isSwitch ? this.renderCheckboxDefault() : this.renderCheckboxSwitch()} `;
+    }
+
+    renderCheckboxDefault(): TemplateResult {
+        const classes = {
+            'vl-checkbox--block': this.block ?? true,
+            'vl-checkbox--error': (this.error || this.isInvalid) ?? true,
+            'vl-checkbox--disabled': this.disabled ?? true,
+        };
+        return html`
+            <label id="label" class="vl-checkbox ${classMap(classes)}" for="checkbox">
+                <input
+                    class="vl-checkbox__toggle"
+                    type="checkbox"
+                    id="checkbox"
+                    name="checkbox"
+                    aria-labelledby=${this.label}
+                    .checked=${this.checked}
+                    ?disabled=${this.disabled}
+                    @click=${this.toggle}
+                />
+                <div class="vl-checkbox__label">
+                    <i class="vl-checkbox__box" aria-hidden="true"></i>
+                    <span>
+                        <slot></slot>
+                    </span>
+                </div>
+            </label>
+        `;
+    }
+
+    renderCheckboxSwitch(): TemplateResult {
+        const classes = {
+            'vl-checkbox--block': this.block ?? true,
+            'vl-checkbox--error': (this.error || this.isInvalid) ?? true,
+            'vl-checkbox--disabled': this.disabled ?? true,
+        };
+        return html`
+            <div class="vl-checkbox--switch__wrapper ${classMap(classes)}">
+                <input
+                    class="vl-checkbox--switch"
+                    type="checkbox"
+                    id="checkbox"
+                    name="checkbox"
+                    aria-labelledby=${this.label}
+                    .checked=${this.checked}
+                    ?disabled=${this.disabled}
+                />
+                <label class="vl-checkbox__label" @click=${() => !this.disabled && this.toggle()}>
+                    <span class="vl-checkbox--switch__label">
+                        <span aria-hidden="true"></span>
+                    </span>
+                    <span>
+                        <slot></slot>
+                    </span>
+                </label>
+            </div>
+        `;
+    }
+
+    toggle(): void {
+        this.checked = !this.checked;
+    }
+
+    get validationTarget(): HTMLInputElement | undefined | null {
+        return this.shadowRoot?.querySelector('input');
+    }
+
+    resetFormControl(): void {
```

Comment:
Zijn er dingen die altijd gereset moeten worden voor elk component dat overerft van FormControl? Indien ja mag je die toevoegen in de resetFormControl van de FormControl.

Reply:
`touched` is iets dat ik nodig had in de checkbox, maar dit is mss niet iets dat we voor alle FormControl's willen; heb wel dan `value` en `initialValue` toegevoegd in een aparte commit (kan dan eventueel gedropt worden); en reset die dan ook conform ik in `vl-checkbox-next` deed + ik reset ook de `error` maar daar ben ik niet zeker van of we dit wel willen

Reply:
zoals besproken, enkel `touched` toegevoegd, maar niet `value` & `initialValue`